File: backend/app/schemas/upstox_credentials.py
"""
Upstox Credentials Pydantic Schemas

Schemas for Upstox API app credential management (Tier 3).
"""
from typing import Optional
from pydantic import BaseModel, Field


class UpstoxCredentialsCreate(BaseModel):
    """Request schema for storing Upstox API app credentials."""
    api_key: str = Field(..., min_length=1, max_length=64, description="Upstox API key")
    api_secret: str = Field(..., min_length=1, max_length=128, description="Upstox API secret")


class UpstoxCredentialsResponse(BaseModel):
    """Response schema for Upstox credentials status (never returns actual secrets)."""
    has_credentials: bool
    api_key: Optional[str] = None
    is_active: bool = False
    last_error: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/services/upstox_credentials.js
/**
 * Upstox Credentials Service
 *
 * API calls for Upstox API app credential management (Tier 3).
 */
import api from './api'

export async function getCredentials() {
  const response = await api.get('/api/upstox-credentials/credentials')
  return response.data
}

export async function storeCredentials(credentials) {
  const response = await api.post('/api/upstox-credentials/credentials', credentials)
  return response.data
}

export async function deleteCredentials() {
  const response = await api.delete('/api/upstox-credentials/credentials')
  return response.data
}

export default { getCredentials, storeCredentials, deleteCredentials }
